packages/hadron-react-components/src/modal-status-message.jsx

```diff
@@ -21,7 +32,8 @@ class ModalStatusMessage extends React.Component {
           <div className="col-md-1">
             <i
               className={`fa fa-${this.props.icon} ${classPrefix}-icon`}
-              aria-hidden="true"></i>
+              aria-hidden="true"
+              onClick={this.onIconClickHandler.bind(this)} />
```

Comment:
Now that we are making this icon actually interactable maybe it makes sense to make it a real `<button>` and remove the `aria-hidden` to make it accessible and more semantically correct, what do you think?

Reply:
Good point. I'm afraid of changing the element as it might cause styling issues when using a button or any other element. I found the similar (maybe not optimal) code in `https://github.com/mongodb-js/hadron-react/blob/master/packages/hadron-react-components/src/info-sprinkle.jsx`. There it is at least without the `aria-hidden` but still an `a` tag.

Removing the `aria-hidden` if and only if there is a click handler should not be an issue. I could also change the cursor to be a `pointer` in that case to make it clear.

@Anemy what do you think?

Reply:
In general Compass could definitely use some accessibility improvements.
I think making it a button when onClick is passed makes sense. It might be tricky with the styling though. If getting the styles to work correctly with `<button>` isn't easy I think it's ok to have it as `i`.
Having `cursor: pointer;` would be nice.

Not for this ticket, but I think down the line we'd like to start using https://github.com/mongodb/leafygreen-ui/tree/master/packages/icon-button where we have icon buttons. (Then we can also start removing some of the font awesome via css classes).

Reply:
The icon with `cursor: pointer` will be fixed after all PRs will be merged. Prepared it in compass already.

Reply:
Yeah, makes sense, if it's too much effort then it's okay to keep it as it is